sysl2/sysl/seqs_test.go

```diff
@@ -398,3 +406,79 @@ func TestDoGenerateSequenceDiagrams(t *testing.T) {
 		})
 	}
 }
+
+func TestDoConstructSequenceDiagramWithGroupingCommandline(t *testing.T) {
+	// Given
+	args := &sdArgs{
+		rootModel: "./tests/",
+		modules:   "call.sysl",
+		output:    "tests/call.png",
+		endpoints: []string{"MobileApp <- Login"},
+		groupbox:  "owner",
+	}
+	var boxPresent bool
+	var err error
+
+	// When
+	boxServer := `box "server" #LightBlue
+	participant _\d
+	participant _\d
+end box`
+	boxClient := `box "client" #LightBlue
+	participant _\d
+	participant _\d
+end box`
+	result := DoConstructSequenceDiagrams(args.rootModel, args.endpointFormat, args.appFormat,
+		args.title, args.output, args.modules, args.endpoints, args.apps, args.blackboxes, args.groupbox)
+
+	// Then
+	boxPresent, err = regexp.MatchString(boxServer, result["tests/call.png"])
+	if err != nil {
+		t.Errorf("Error compiling regular expression")
+	}
+	assert.True(t, boxPresent)
+	boxPresent, err = regexp.MatchString(boxClient, result["tests/call.png"])
+	if err != nil {
+		t.Errorf("Error compiling regular expression")
+	}
+	assert.True(t, boxPresent)
+	cnt := strings.Count(result["tests/call.png"], "participant")
+	assert.True(t, cnt == 4)
+}
+
+func TestDoConstructSequenceDiagramWithGroupingSysl(t *testing.T) {
+	// Given
+	args := &sdArgs{
+		rootModel: "./tests/",
+		modules:   "groupby.sysl",
+		output:    "%(epname).png",
+		endpoints: []string{"SEQ-One"},
+		apps:      []string{"Project :: Sequences"},
+	}
+	var boxPresent bool
+	var err error
+
+	// When
+	boxOnpremise := `box "onpremise" #LightBlue
+	participant _\d
+	participant _\d
+end box`
+	boxCloud := `box "cloud" #LightBlue
+	participant _\d
+	participant _\d
+end box`
+	result := DoConstructSequenceDiagrams(args.rootModel, args.endpointFormat, args.appFormat,
+		args.title, args.output, args.modules, args.endpoints, args.apps, args.blackboxes, args.groupbox)
+
+	// Then
+	boxPresent, err = regexp.MatchString(boxOnpremise, result["SEQ-One.png"])
+	if err != nil {
+		t.Errorf("Error compiling regular expression")
+	}
+	assert.True(t, boxPresent)
+	boxPresent, err = regexp.MatchString(boxCloud, result["SEQ-One.png"])
+	if err != nil {
+		t.Errorf("Error compiling regular expression")
+	}
+	assert.True(t, boxPresent)
+}
```

Comment:
test for Seq-Two.png having only one app in one group.

Reply:
Added